include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -37,6 +37,44 @@ namespace ExaDG
 {
 using namespace dealii;
 
+template<typename Operator>
+std::unique_ptr<MPI_Comm, void (*)(MPI_Comm *)>
+create_subcommunicator(Operator const & op)
+{
+  unsigned int n_locally_owned_cells = 0;
+  auto const & dof_handler           = op.get_matrix_free().get_dof_handler();
```

Comment:
`MatrixFree` can have multiple `DofHandler`s, so I think this is error prone (`op.get_dof_index()`!). I my opinion it would make much more sense to keep the `Operator` out of the interface and use `DofHandler` instead.

Reply:
With this change, this function should only depend on `dealii::DofHandler`, and MPI --> shift to deal.II lateron.

Reply:
Fixed in b643a551